generic_server.cc

```diff
@@ -167,16 +168,28 @@ future<> server::shutdown() {
 }
 
 future<>
-server::listen(socket_address addr, std::shared_ptr<seastar::tls::credentials_builder> builder, bool is_shard_aware, bool keepalive, std::optional<file_permissions> unix_domain_socket_permissions) {
-    shared_ptr<seastar::tls::server_credentials> creds = nullptr;
+server::listen(socket_address addr, std::shared_ptr<seastar::tls::credentials_builder> builder, bool is_shard_aware, bool keepalive, std::optional<file_permissions> unix_domain_socket_permissions, std::function<server&()> get_shard_instance) {
```

Comment:
Is this not already covered by peering_sharded_service?

Reply:
generic_server::server is not a peering_sharded_service. It is in fact not even sharded at all. Subclasses are, thus they can provide this info in the callback.